ledger/participant-integration-api/src/main/scala/platform/store/dao/events/IdPageSizing.scala

```diff
@@ -18,46 +18,71 @@ case class IdPageSizing(
-        ./(
-          idPageBufferSize + 1
-        ) // for each filter we need one page fetched for merge sorting, and additional pages might reside in the buffer
-        ./(filterSize)
-    if (calculatedMaxIdPageSize < LowestIdPageSize) {
+    val calculated = calculateMaxNumOfIdsPerPage(
+      workingMemoryInBytesForIdPages = workingMemoryInBytesForIdPages,
+      numOfDecomposedFilters = numOfDecomposedFilters,
+      numOfPagesInIdPageBuffer = numOfPagesInIdPageBuffer,
+    )
+    // maxNumberOfIdsPerIdPage can override this if it is smaller
+    val minIdPageSize = Math.min(10, maxIdPageSize)
+    // maxNumberOfIdsPerIdPage can override this if it is smaller
+    val recommendedIdPageSize = Math.min(NumOfBtreeLeafPageEntriesApprox, maxIdPageSize)
+    // An id occupies 8 bytes
+    if (calculated < minIdPageSize) {
       logger.warn(
-        s"Calculated maximum ID page size supporting API stream memory limits [$calculatedMaxIdPageSize] is too low: $LowestIdPageSize is used instead. Warning: API stream memory limits not respected. Warning: Dangerously low maximum ID page size can cause poor streaming performance. Filter size [$filterSize] too large?"
+        s"Calculated maximum ID page size supporting API stream memory limits [$calculated] is too low: $minIdPageSize is used instead. " +
+          s"Warning: API stream memory limits not respected. Warning: Dangerously low maximum ID page size can cause poor streaming performance. " +
+          s"Filter size [$numOfDecomposedFilters] too large?"
       )
-      IdPageSizing(LowestIdPageSize, LowestIdPageSize)
-    } else if (calculatedMaxIdPageSize < RecommendedMinIdPageSize) {
+      IdPageSizing(minIdPageSize, minIdPageSize)
+    } else if (calculated < recommendedIdPageSize) {
       logger.warn(
-        s"Calculated maximum ID page size supporting API stream memory limits [$calculatedMaxIdPageSize] is very low. Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$filterSize] too large?"
+        s"Calculated maximum ID page size supporting API stream memory limits [$calculated] is very low. " +
+          s"Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$numOfDecomposedFilters] too large?"
       )
-      IdPageSizing(calculatedMaxIdPageSize, calculatedMaxIdPageSize)
-    } else if (calculatedMaxIdPageSize < maxIdPageSize) {
+      IdPageSizing(calculated, calculated)
+    } else if (calculated < maxIdPageSize) {
       logger.info(
-        s"Calculated maximum ID page size supporting API stream memory limits [$calculatedMaxIdPageSize] is low. Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$filterSize] too large?"
+        s"Calculated maximum ID page size supporting API stream memory limits [$calculated] is low. " +
+          s"Warning: Low maximum ID page size can cause poor streaming performance. Filter size [$numOfDecomposedFilters] too large?"
```

Comment:
There is a `Warning:` logged in info level. It seems that there is not a situation where the calculation is correct.

Reply:
i.e. there is a complaint for each branch of the if-else statement.

Reply:
The last one is without a warning.
I intentionally haven't changed the wording of these messages (as e.g. someone might rely on them to diagnose problem). I only refactored the surrounding code